script/publish.ts

```diff
@@ -163,6 +165,37 @@ function upload(assetName: string, assetPath: string) {
   })
 }
 
+function getAzureBlobService(): Azure.BlobService | null {
+  if (
+    process.env.AZURE_STORAGE_ACCOUNT === undefined ||
+    process.env.AZURE_STORAGE_ACCESS_KEY === undefined ||
+    process.env.AZURE_BLOB_CONTAINER === undefined
+  ) {
+    console.error('Invalid azure storage credentials')
+    return null
+  }
+
+  const blobService = Azure.createBlobService(
+    process.env.AZURE_STORAGE_ACCOUNT,
+    process.env.AZURE_STORAGE_ACCESS_KEY
+  )
+
+  blobService.createContainerIfNotExists(
+    process.env.AZURE_BLOB_CONTAINER,
+    {
+      publicAccessLevel: 'blob',
+    },
+    error => {
+      if (error !== undefined) {
+        console.error(
+          `Unable to ensure azure blob container - ${process.env.AZURE_BLOB_CONTAINER}`
+        )
+      }
+    }
+  )
```

Comment:
Do we need to handle this asynchronously? According to https://azure.github.io/azure-storage-node/BlobService.html#createContainerIfNotExists__anchor the last parameter is a callback saying whether the result was successful or not.

From what I'm reading, sounds like we don't care about the result, just logging the error to give us a clue but not halting the publish process. Is that correct?

Reply:
Yes. I wasn't super convinced that this was all that valuable to our solution (based my solution on some examples and they all  had it). Because really if this azure container didn't exist, it would fail on the upload anyways... and likely there would be bigger problems.. Would you prefer just to remove this error logging?

Reply:
Nope, I think the error is interesting! Maybe posting it as a warning?

And also a comment stating that we're intentionally not waiting for the operation to complete.

Reply:
Decided since the next call would fail if this one did.. probably didn't make sense to go ahead with it. Refactored this to a promise so we don't bother attempting upload if there is no container to upload to.